src/magic_folder/web.py

```diff
@@ -428,6 +432,24 @@ def all_folder_configs():
             in all_folder_configs()
         })
 
+    @app.route("/magic-folder/<string:folder_name>/file-status", methods=['GET'])
+    def folder_file_status(self, request, folder_name):
+        """
+        Render status information for every file in a given folder
+        """
+        _application_json(request)  # set reply headers
+        folder_config = self._global_config.get_magic_folder(folder_name)
+
+        return json.dumps([
+            {
+                "name": name,
+                "mtime": ns_to_seconds(ps.mtime_ns),
```

Comment:
I was considering suggesting that this should be a method on `PathState`, but I guess maybe it would be to similarly named to `.mtime_ns`, and apt to cause confusion or error. (Certainly it shouldn't be a property for that reason)